apps/astarte_realm_management_api/test/astarte_realm_management_api/interfaces/interfaces_test.exs

```diff
@@ -150,8 +154,14 @@ defmodule Astarte.RealmManagement.API.InterfacesTest do
   describe "interface update" do
     @describetag :update
 
-    setup %{realm: realm} do
-      {:ok, %Interface{}} = Interfaces.create_interface(realm, @valid_attrs)
+    setup %{realm: realm, astarte_instance_id: astarte_instance_id} do
+      {:ok, interface} = create_interface(realm, @valid_attrs)
+
+      on_exit(fn ->
+        Database.setup_database_access(astarte_instance_id)
+        Engine.delete_interface(realm, interface.name, interface.major_version)
```

Comment:
`Engine.delete_interface/3` checks for various things when deleting an interface (namely the version number, non-zero interfaces cannot be deleted), I suggest to use `RealmManagement.Queries.delete_interface/3` as it deletes the given interface bypassing the checks